.internal-ci/util/metadata.sh

```diff
@@ -25,66 +17,107 @@ is_set()
   fi
 }
 
+filter()
```

Comment:
nit: (pretty nitty, I would probably ignore for this PR and if there is anything of value consider in the future.)
1. It may be my notoriously poor vocabulary or general lack of domain knowledge, but I think of `filter` as "removing" and/or "allowing" certain items to proceed.   Like an air filter or oil filter.  Would `normalize` be a better term here?  I know normalize can be a bit loaded and general at times.
2. often in programing languages filter is accompanied by a function or functor, would it be better to more explicitly name it `filter_git_ref` or similar?

again these are pretty nitty so ignore as you see fit

Reply:
Good points, naming things is hard :grin: